astropy/table/operations.py

```diff
@@ -808,7 +810,7 @@ def _join(left, right, keys=None, join_type='inner',
 
             out[out_name] = col_cls.info.new_like(cols, n_out, metadata_conflicts, out_name)
 
-            if issubclass(col_cls, Column):
+            if not NUMPY_LT_1_17 or issubclass(col_cls, (Column, Time)):
```

Comment:
Is the NUMPY_L_1_17 stuff still useful?

Reply:
yes, I'll definitely keep that in here. I'm not sure about the usefulness of the tests though?

Reply:
Well I don't know enough about the time series code to know.  But maybe it's a good idea as a regression to ensure future changes in Table or TimeSeries don't break things.